lib/MusicBrainz/Server/Edit/Artist/Merge.pm

```diff
@@ -51,11 +56,61 @@ sub do_merge
 {
     my $self = shift;
 
-    $self->c->model('Artist')->merge(
-        $self->new_entity->{id},
-        [ $self->_old_ids ],
+    my $new_id = $self->new_entity->{id};
+    my @old_ids = $self->_old_ids;
+    my $all_ids = [$new_id, @old_ids];
+
+    my (undef, $dropped_columns) = $self->c->model('Artist')->merge(
+        $new_id,
+        \@old_ids,
         rename => $self->data->{rename}
     );
+
+    if ($dropped_columns->{type}) {
+        my $dropped_type =
+            $self->c->model('ArtistType')->get_by_id($dropped_columns->{type});
+
+        $self->c->model('EditNote')->add_note(
+            $self->id => {
+                editor_id => $EDITOR_MODBOT,
+                text => localized_note(
+                    'The “{artist_type}” type has been dropped because ' .
+                    'one of the artists has a gender.',
+                    vars => {
+                        artist_type => localized_note(
+                            $dropped_type->name,
+                            function => 'lp',
+                            domain => 'attributes',
+                            args => ['artist_type'],
+                        ),
+                    },
+                ),
+            },
+        );
+    }
+
+    if ($dropped_columns->{gender}) {
+        my $dropped_gender =
+            $self->c->model('Gender')->get_by_id($dropped_columns->{gender});
+
+        $self->c->model('EditNote')->add_note(
+            $self->id => {
+                editor_id => $EDITOR_MODBOT,
+                text => localized_note(
+                    'The “{gender}” gender has been dropped because ' .
+                    'one of the artists is a group.',
```

Comment:
'The “{gender}” gender has been dropped because the destination artist is a group.'? I mean, if a separate artist is a group but the destination artist is already a person, I'd expect to still assign the gender to it. I guess that wording doesn't work for the case from the test where the type is also being dropped though...